### PR TITLE
move sparsity related discussion to another section.

### PR DESCRIPTION
**Pull request by Ankush-Chander**
Many thanks for this comprehensive guide on embeddings. 

My understanding behind proposed change: 
Sparsity related discussion at the beginning of section `3.2.2 TF-IDF` gives an impression that we solve sparsity issue by moving from one-hot encoding to TF-IDF. However TF-IDF encoding doesn"t change the number of zero cells we have in one-hot encoding, it only effects the weightage given to the non-zero cells, thereby leaving the sparsity of the one-hot matrix unchanged. 

Proposed change: 
1. Elicit advantage of TF-IDF over one-hot encoding in section `3.2.2 TF-IDF`. 
2. Move sparsity discussion to section `3.2.3` where matrix factorisation is discussed.

### CODE REVIEW

**Comment by micaleel** [comment]
@Ankush-Chander, your definition of sparsity based on the size of embeddings is debatable . When comparing two embeddings of equal size, the one with fewer empty or zero values is considered dense.

**Comment by Ankush-Chander** [comment]
Hi @micaleel 
Sorry for not being more explicit before. My interpretation of sparsity is exactly as you just mentioned. 
Number of cells in term-document matrix that have value 0.

File: embeddings.tex [hunk 1]
```diff
@@ -1096,25 +1096,7 @@ \subsubsection*{Embeddings as larger feature inputs}
...
-]{python}
-sparse_vector = [1,0,0,0,0,0,0,0,0,0]
-dense_vector = [1,2,2,3,0,4,5,8,8,5]
-\end{minted}
-\caption{Two types of vectors in text processing}
-\end{figure}
-
-
-Sparse vectors result in a number of problems, among these \textbf{cold start}---the idea that we don't know to recommend items that haven't been interacted with, or for users who are new. What we'd like, instead, is to create dense vectors, which will  give us more information about the data, the most important of which is accounting for the weight of a given word in proportion to other words. This is where we leave one-hot encodings and move into approaches that are meant to solve for this sparsity. Dense vectors are just vectors that have mostly non-zero values. We call these dense representations dynamic representations \citep{Wang2020FromST}.
+One-hot encoding just deals with presence and absence of a term in a document. However it is preferable that we differentiate the importance/weightage of terms in context of the document collection as a whole instead of just the individual document.
```
Comment (veekaybee) [inline embeddings.tex:1099]: I'd reword this as, 

"One-hot encoding just deals with presence and absence of a single term in a single document. However, when we have large amounts of data, we'd like to consider the weights of each term in relation to all the other terms in a collection of documents."

**Review by veekaybee — COMMENTED** [review]
Left one small change for clarification, once that's done this is good to go. Thanks for opening this!
